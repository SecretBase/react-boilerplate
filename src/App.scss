@import './color-palette';

.title {
  color: red;
}

.title--blue {
  color: blue;
}
