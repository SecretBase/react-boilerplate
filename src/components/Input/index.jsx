// @flow
import styled from 'styled-components'

const Input = styled.input`
  border: 1px solid #ddd;
  width: 980px;
  margin: 16px auto;
  box-sizing: border-box;
  display: block;
  font-size: 24px;
  padding: 8px 16px;
`

export default Input
