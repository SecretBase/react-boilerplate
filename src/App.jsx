// @flow
import React, { PureComponent } from 'react'

import MoiveList from './components/MoiveList'
import type {Moives} from './components/MoiveList'
import Input from './components/Input'

type Props = {}
type State = {
  moives: Moives,
  query: string
}

export const getMoives = (): Moives => [
  {
    name: 'Spider Man',
    emoji: '🕷 👦🏻'
  },
  {
    name: 'Frozen',
    emoji: '❄️ 👗'
  },
  {
    name: 'Cars 3',
    emoji: '🚘 🚘 🚘'
  },
  {
    name: 'Frozen 2',
    emoji: '❄️ 👗 ✌️'
  },
  {
    name: 'Harry Potter',
    emoji: '👦🏻 👓 ⚡️'
  }
]

class App extends PureComponent<Props, State> {
  state = {
    moives: getMoives(),
    query: ''
  }

  filterMovies = () => {
    this.setState(() => {
      return {
        moives: getMoives()
      }
    })
  }

  onChange = (event: Event) => {
    const {target} = event
    if (!(target instanceof global.HTMLInputElement)) return
    this.setState(() => {
      return {
        query: target.value,
        moives: getMoives()
          .filter(moive => moive.name
            .toLowerCase()
            .includes(target.value.toLowerCase()))
      }
    })
  }

  render () {
    return [
      <Input
        onChange={this.onChange}
        placeholder='Search'
        key='input'
        value={this.state.query}
      />,
      <MoiveList
        moives={this.state.moives}
        key='moiveList'
      />
    ]
  }
}

export default App
