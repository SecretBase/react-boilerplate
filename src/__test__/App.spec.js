import React from 'react'
import {mount} from 'enzyme'
import renderer from 'react-test-renderer'

import App, {getMoives} from '../App'
import Input from '../components/Input'

describe('App Component', () => {
  it('should correct rendered.', () => {
    const component = renderer.create(<App />)

    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should return a list of movies', () => {
    expect(getMoives()).toMatchSnapshot()
  })

  it('should render a only Cars 3', () => {
    const wrapper = mount(
      <App />
    )

    wrapper.setState((prev) => {
      return {
        ...prev,
        query: 'Cars'
      }
    })

    const input = wrapper.find(Input)

    input.simulate('change')

    expect(wrapper).toMatchSnapshot()
  })
})
