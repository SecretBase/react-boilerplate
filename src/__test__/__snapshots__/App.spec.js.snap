// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should correct rendered. 1`] = `
Array [
  <input
    className="sc-bxivhb xxaXh"
    onChange={[Function]}
    placeholder="Search"
    value=""
  />,
  <ul
    className="sc-htpNat flqtFs"
  >
    <li
      className="sc-bdVaJa kmQhNa"
    >
       
      🕷 👦🏻
       
      <span
        className="sc-bwzfXH hXzJNw"
      >
        Spider Man
      </span>
       
    </li>
    <li
      className="sc-bdVaJa kmQhNa"
    >
       
      ❄️ 👗
       
      <span
        className="sc-bwzfXH hXzJNw"
      >
        Frozen
      </span>
       
    </li>
    <li
      className="sc-bdVaJa kmQhNa"
    >
       
      🚘 🚘 🚘
       
      <span
        className="sc-bwzfXH hXzJNw"
      >
        Cars 3
      </span>
       
    </li>
    <li
      className="sc-bdVaJa kmQhNa"
    >
       
      👦🏻 👓 ⚡️
       
      <span
        className="sc-bwzfXH hXzJNw"
      >
        Harry Potter
      </span>
       
    </li>
  </ul>,
]
`;

exports[`App Component should render a only Cars 3 1`] = `
<App>
  <styled.input
    key="input"
    onChange={[Function]}
    placeholder="Search"
    value="Cars"
  >
    <input
      className="sc-bxivhb xxaXh"
      onChange={[Function]}
      placeholder="Search"
      value="Cars"
    />
  </styled.input>
  <MoiveList
    key="moiveList"
    moives={
      Array [
        Object {
          "emoji": "🚘 🚘 🚘",
          "name": "Cars 3",
        },
      ]
    }
  >
    <styled.ul>
      <ul
        className="sc-htpNat flqtFs"
      >
        <MoiveItem
          emoji="🚘 🚘 🚘"
          key="Cars 3"
          name="Cars 3"
        >
          <styled.li>
            <li
              className="sc-bdVaJa kmQhNa"
            >
               
              🚘 🚘 🚘
               
              <styled.span>
                <span
                  className="sc-bwzfXH hXzJNw"
                >
                  Cars 3
                </span>
              </styled.span>
               
            </li>
          </styled.li>
        </MoiveItem>
      </ul>
    </styled.ul>
  </MoiveList>
</App>
`;

exports[`App Component should return a list of moives 1`] = `
Array [
  Object {
    "emoji": "🕷 👦🏻",
    "name": "Spider Man",
  },
  Object {
    "emoji": "❄️ 👗",
    "name": "Frozen",
  },
  Object {
    "emoji": "🚘 🚘 🚘",
    "name": "Cars 3",
  },
  Object {
    "emoji": "👦🏻 👓 ⚡️",
    "name": "Harry Potter",
  },
]
`;
